lib/dry/schema/dsl.rb

```diff
@@ -84,6 +103,30 @@ def set_type(name, spec)
         types[name] = resolve_type(spec).meta(omittable: true)
       end
 
+      def key_map(types = self.types)
+        keys = types.keys.each_with_object([]) { |a, e|
+          e << key_value(a, types[a])
+        }
```

Comment:
Having single-letter block args here certainly impeded my understanding of this as someone coming in relatively fresh.

Could we change them to `|key_name, arr|` instead? Or something similar?